### ISSUE TITLE
"@require-wp-4.7.3" Behat tag doesn't work when `WP_VERSION=latest`

### ISSUE THREAD

**Issue by danielbachhuber** [wp-cli/wp-cli#3956]
> Note though that the "require-wp-4.7.3" tests are being skipped, due to the WP_VERSION being "latest" or "trunk" issue.
>
> Perhaps a nice way to do deal with it (rather than the dummy WP_VERSION hack) would be to ignore "require-wp" in "behat-tags.php" if the WP_VERSION is "latest" or "trunk", eg [https://github.com/gitlost/wp-cli/blob/behat_tags_wp_version/ci/behat-tags.php#L17](https://github.com/gitlost/wp-cli/blob/behat_tags_wp_version/ci/behat-tags.php#L17).
> 
> I think this makes sense as "latest" or "trunk" must satisfy any WP version requirement...

From https://github.com/wp-cli/media-command/pull/4#issuecomment-290563489

**Comment by gitlost** [comment]
I suppose one could resolve "latest" at least by using https://api.wordpress.org/core/version-check (there doesn't seem to be an equivalent for "trunk"?) but it seems like a lot of work and complication for little gain.

**Comment by schlessera** [comment]
Maybe we could just extract the version from `latest` and `trunk` and do at least a numeric comparison.

Otherwise, it would mean that we always have trouble testing "cutting-edge" functionality until it is not cutting-edge anymore.

**Comment by gitlost** [comment]
> Maybe we could just extract the version from latest and trunk and do at least a numeric comparison.

How to do that though? And is it worth it?!

> Otherwise, it would mean that we always have trouble testing "cutting-edge" functionality until it is not cutting-edge anymore.

Perhaps using a tag like "wp-require-trunk" would meet this use case? See https://github.com/gitlost/wp-cli/pull/1 for an implementation of this. The build of which threw up a couple of things:

* Alot more tests get run!
* A test failure in "taxonomy.feature" due to a data change in the `link_category` taxonomy in WP 4.4 (see https://core.trac.wordpress.org/ticket/34284).

**Comment by danielbachhuber** [comment]
> How to do that though?

We can run:

```
wp core download --version=$WP_VERSION --path=/wp-cli-test core-download-cache
wp core version --path=/wp-cli-test core-download-cache
```

> And is it worth it?!

Behat makes the HTTP request anyway, so we're only moving it earlier in the process.

**Comment by gitlost** [comment]
Ah I see, very good, so that's the way to do it then.

**Comment by danielbachhuber** [comment]
@gitlost I decided to go with your original approach (ignoring `@require-wp-(.+)` entirely for 'latest') because it turned out to be more elegant.

**Comment by schlessera** [comment]
The above approach actually breaks tests in a different use case right now: https://travis-ci.org/wp-cli/entity-command/jobs/366509094#L589

The problem: The test does `@require-wp-5.0` because it tests code that is only available from 5.0 onwards. However, for all tests run on WordPress `latest`, the tags do not get properly created, and so that test runs nevertheless and throws an error.

We need to actually resolve what version we're using when stating `latest` to be able to have tests for upcoming versions of WordPress not break.

**Comment by danielbachhuber** [comment]
@schlessera Can you please open a new issue referencing this issue, instead of re-opening an issue assigned to an old milestone?
